### PR TITLE
Add user management routes

### PR DESCRIPTION
**Pull request by fharding1**
Add /users for getting all users, and /users/{username} for updating a user.

### CODE REVIEW

**Comment by calebeby** [comment]
Because frank wants to send extra data to the frontend that will never be displayed (that's what he said)

**Comment by BrendanBurkhart** [comment]
That's a security breach though, if I really wanted to I could use that info to crack a user's password fairly easily.

File: internal/store/user/postgres/postgres.go [hunk 1]
```diff
@@ -23,9 +29,28 @@ func (s *Service) Get(username string) (u user.User, err error) {
...
-func (s *Service) Create(u user.User) error {
-	_, err := s.db.Exec("INSERT INTO users VALUES ($1, $2, $3)", u.Username, u.HashedPassword, u.IsAdmin)
+// GetUsers gets all users in the postgresql database.
+func (s *Service) GetUsers() ([]user.User, error) {
+	rows, err := s.db.Query("SELECT username, hashedPassword, isAdmin from users")
+	if err != nil {
+		return []user.User{}, err
+	}
+
+	var users []user.User
```
Comment (BrendanBurkhart) [inline internal/store/user/postgres/postgres.go]: I think we need a `defer rows.Close()` here.

File: internal/server/users.go [hunk 2]
```diff
@@ -104,14 +125,54 @@ func (s *Server) createUserHandler(w http.ResponseWriter, r *http.Request) {
...
+
+	authenticatedUser, uOk := r.Context().Value(keyUsernameCtx).(string)
+	authenticatedIsAdmin, aOk := r.Context().Value(keyIsAdminCtx).(bool)
+
+	if !uOk || !aOk {
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	var reqUser struct {
```
Comment (BrendanBurkhart) [inline internal/server/users.go]: We don't need this anymore do we?

**Pull request merged**